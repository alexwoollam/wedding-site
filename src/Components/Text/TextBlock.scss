.text-block{
  line-height: 2rem;
  color: #000;
  margin: 0 auto;
  padding: 0;
  max-width: 550px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 30px;
}

a {
  color: #67805a;
  text-decoration: none;
  text-align: center;
  display: block;
  justify-content: center;
}