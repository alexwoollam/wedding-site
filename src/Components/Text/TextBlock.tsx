import React from 'react';

function TextBlock( props ) {
  const { children } = props;
  return (
    <p 
    className="text-block text-gray-600 font-sans text-base text-center" 
    data-testid="text-block"
    >
        { children }
    </p>
  );
}

export default TextBlock;