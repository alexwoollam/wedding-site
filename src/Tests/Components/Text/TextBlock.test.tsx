import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import TextBlock from '../../../Components/Text/TextBlock';

describe('<TextBlock/> component', () => {
  test('renders keep me updated form.', () => {
      render(<TextBlock>text block</TextBlock>);
    const linkElement = screen.getByText('text block');
    expect(linkElement).toBeInTheDocument();
  });

  test('If no props are sent, do an early return.', () => {
      render(<TextBlock></TextBlock>);
      const linkElement = screen.queryByTestId('text-block');
      expect(linkElement).not.toBeInTheDocument();
  });
  it('renders correctly', () => {
    const tree = renderer
        .create(<TextBlock>text block</TextBlock>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
