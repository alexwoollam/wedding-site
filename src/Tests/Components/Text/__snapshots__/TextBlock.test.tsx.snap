// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextBlock/> component renders correctly 1`] = `
<p
  className="text-block text-gray-600 font-sans text-base text-center"
  data-testid="text-block"
>
  text block
</p>
`;
